src/ui/public/modals/confirm_modal_promise.js

```diff
@@ -0,0 +1,31 @@
+import uiModules from 'ui/modules';
+import 'ui/modals';
+
+const module = uiModules.get('kibana');
+
+/**
+ * @typedef {Object} PromisifiedConfirmOptions
+ * @property {String=} confirmButtonText
+ * @property {String=} cancelButtonText
+ */
+
+/**
+ * A "promisified" version of ConfirmModal that binds onCancel and onConfirm to
+ * Resolve and Reject methods.
+ */
+module.factory('confirmModalPromise', function (Promise, confirmModal) {
+  /**
+   * @param {String} message
+   * @param {PromisifiedConfirmOptions} customOptions
+   */
+  return (message, customOptions) => new Promise((resolve, reject) => {
+    const defaultOptions = {
+      onConfirm: resolve,
+      onCancel: reject,
+      confirmButtonText: 'Okay',
```

Comment:
Can we remove this default from both here and `confirm_modal.js`? I think I accidentally let it slip by in the original PR, sorry about that.